CopilotKit/packages/react-core/src/hooks/use-copilot-chat.ts

```
@@ -80,440 +101,75 @@
-    interrupt,
-  };
-}
+  }, [hasPublicApiKey]); // Removed setBannerError dependency
 
-// store `value` in a ref and update
-// it whenever it changes.
-function useUpdatedRef<T>(value: T) {
-  const ref = useRef(value);
+  // Use internal implementation if publicApiKey is available
+  if (hasPublicApiKey) {
+    return useCopilotChatInternal(options);
```

Biome (2.1.2)

[error] This hook is being called conditionally, but all hooks must be called in the exact same order in every component render.

For React to preserve state between calls, hooks needs to be called unconditionally and always in the same order.
See https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

(lint/correctness/useHookAtTopLevel)